File: Java/easy/Exercism/ReverseString.java
package easy.Exercism;

class ReverseString {

    // Reverse a string

    // For example:
    // input: "cool"
    // output: "looc"
    String reverse(String inputString) {
        StringBuffer sbf = new StringBuffer(inputString);
        return sbf.reverse().toString();
    }
  
}

File: Golang/easy/Exercism/reverse_string.go
package reverse

// Reverse a string

// For example:
// input: "cool"
// output: "looc"

func Reverse(input string) string {
	// 4123 ns/op
	// reversed := ""
	// for _, v := range input {
	// 	reversed = string(v) + reversed
	// }
	// return reversed

	//1167 ns/op
	if len(input) == 0 {
		return ""
	}
	runes := []rune(input)

	reversed := make([]rune, 0, cap(runes))
	for i := len(runes) - 1; i >= 0; i-- {
		reversed = append(reversed, []rune(input)[i])
	}

	return string(reversed)
}
